Sources/NIOPosix/PendingWritesManager.swift

```diff
@@ -329,6 +440,81 @@ final class PendingStreamWritesManager: PendingWritesManager {
         return self.state.currentBestWriteMechanism
     }
 
+#if SWIFTNIO_USE_IO_URING && os(Linux)
+
+    func triggerAppropriateAsyncWriteOperations(scalarBufferAsyncWriteOperation: (UnsafeRawBufferPointer) throws -> Void,
```

Comment:
This is just one example, but this PR has a lot of API surface guarded behind availability of IOUring. I'd like us to minimise this as much as possible, so I think we can do a few things to produce fewer compile conditions.

Firstly, let's make as much API available on all platforms as possible. If the type signature is cross-platform, we should include it. In this case, there's nothing in this function that requires liburing, so we can safely just make this available unconditionally. 

Secondly, where the _implementation_ can also be written safely on all platforms, we should keep it! The goal is to funnel down to the places where we actually interact with io_uring. At those levels, we should have thin layer that calls into an underlying implementation if it's present, or that crashes if called on the wrong platform.

This will hopefully minimise the amount of `#if` we have to scatter around the codebase and make the whole thing substantively easier to read.

Reply:
The idea was do not let code which will be unused on the particular platform even compile.
But if you think it is fine I will remove those conditional compilations in such places.

Reply:
I think it's fine.